server/src/main/java/com/cloud/template/HypervisorTemplateAdapter.java

```diff
@@ -359,6 +361,16 @@ public List<TemplateOrVolumePostUploadCommand> doInTransaction(TransactionStatus
                     throw new CloudRuntimeException("Unable to persist the template " + profile.getTemplate());
                 }
 
+                // Set Event Details for Template/ISO Upload
+                String eventType = template.getFormat().equals(ImageFormat.ISO) ? "Iso" : "Template";
+                String eventResourceId = template.getUuid();
+                CallContext.current().setEventDetails(String.format("%s Id: %s", eventType, eventResourceId));
+                CallContext.current().putContextParameter(eventType.equals("Iso") ? eventType : VirtualMachineTemplate.class, eventResourceId);
+                if (template.getFormat().equals(ImageFormat.ISO)) {
```

Comment:
Is this required to be specific to only ISOs?

Reply:
Yes, otherwise the event resource was not being obtained